### PR TITLE
Coverity 06

### PR DESCRIPTION
**Pull request by andrewbird**
More unchecked returns

### CODE REVIEW

File: src/base/misc/ioctl.c [hunk 1]
```diff
@@ -148,16 +148,33 @@ void
...
     if ((new_fd+1) > numselectfd) numselectfd = new_fd+1;
     if (numselectfd > MAX_FD) {
 	error("Too many IO fds used.\n");
 	leavedos(76);
     }
     flags = fcntl(new_fd, F_GETFL);
-    fcntl(new_fd, F_SETOWN, getpid());
-    fcntl(new_fd, F_SETFL, flags | O_ASYNC);
-    fcntl(new_fd, F_SETFD, FD_CLOEXEC);
+    if (flags == -1) {
```
Comment (stsp) [inline src/base/misc/ioctl.c:159]: You can just do
if (!fcntl() || !fcntl() || !fcntl()) error();
to noe write too much.

Comment (andrewbird) [reply src/base/misc/ioctl.c:159]: okay

Comment (andrewbird) [reply src/base/misc/ioctl.c]: Is this better?
```diff
-    flags = fcntl(new_fd, F_GETFL);
-    fcntl(new_fd, F_SETOWN, getpid());
-    fcntl(new_fd, F_SETFL, flags | O_ASYNC);
-    fcntl(new_fd, F_SETFD, FD_CLOEXEC);
+
+    if ((flags = fcntl(new_fd, F_GETFL)) == -1 ||
+            fcntl(new_fd, F_SETOWN, getpid()) == -1 ||
+            fcntl(new_fd, F_SETFL, flags | O_ASYNC) == -1 ||
+            fcntl(new_fd, F_SETFD, FD_CLOEXEC) == -1) {
+       error("add_to_io_select_new: Fcntl failed\n");
+       leavedos(76);
+    }
+
```

**Comment by stsp** [comment]
Thanks.

**Pull request merged**